src/libm/sleefsimddp.c

```diff
@@ -1844,25 +1916,241 @@ EXPORT CONST vdouble xfmod(vdouble x, vdouble y) {
+  z = ddmul_vd2_vd2_vd(z, t);
+
+  clc = vsel_vd2_vo_vd2_vd2(o2, y, z);
+  
+  clld = vsel_vd2_vo_vd2_vd2(o2, ddadd2_vd2_vd2_vd(ddmul_vd2_vd_vd(u, t), vcast_vd_d(1)), clld);
+  
+  y = clln;
+
+  clc = vsel_vd2_vo_vd2_vd2(otiny, vcast_vd2_d_d(83.1776616671934334590333, 3.67103459631568507221878e-15), // log(2^120)
+			    vsel_vd2_vo_vd2_vd2(oref, ddadd2_vd2_vd2_vd2(vcast_vd2_d_d(1.1447298858494001639, 1.026595116270782638e-17), ddneg_vd2_vd2(clc)), clc)); // log(M_PI)
+  clln = vsel_vd2_vo_vd2_vd2(otiny, vcast_vd2_d_d(1, 0), vsel_vd2_vo_vd2_vd2(oref, clln, clld));
+
+  if (!vtestallones_i_vo64(vnot_vo64_vo64(oref))) {
+    t = vsub_vd_vd_vd(a, vmul_vd_vd_vd(vcast_vd_d(1LL << 28), vcast_vd_vi(vtruncate_vi_vd(vmul_vd_vd_vd(a, vcast_vd_d(1.0 / (1LL << 28)))))));
+    x = ddmul_vd2_vd2_vd2(clld, sinpik(t));
+  }
+  
+  clld = vsel_vd2_vo_vd2_vd2(otiny, vcast_vd2_vd_vd(vmul_vd_vd_vd(a, vcast_vd_d((1LL << 60)*(double)(1LL << 60))), vcast_vd_d(0)),
+			     vsel_vd2_vo_vd2_vd2(oref, x, y));
+
+  dd2 ret = { clc, dddiv_vd2_vd2_vd2(clln, clld) };
+
+  return ret;
+}
+
+EXPORT CONST vdouble xtgamma_u1(vdouble a) {
+  dd2 d = gammak(a);
+  vdouble2 y = ddmul_vd2_vd2_vd2(expk2(d.a), d.b);
+  vdouble r = vadd_vd_vd_vd(y.x, y.y);
+  vopmask o;
+
+  o = vor_vo_vo_vo(vor_vo_vo_vo(veq_vo_vd_vd(a, vcast_vd_d(-INFINITY)),
+				vand_vo_vo_vo(vlt_vo_vd_vd(a, vcast_vd_d(0)), visint_vo_vd(a))),
+		   vand_vo_vo_vo(vand_vo_vo_vo(visnumber_vo_vd(a), vlt_vo_vd_vd(a, vcast_vd_d(0))), visnan_vo_vd(r)));
+  r = vsel_vd_vo_vd_vd(o, vcast_vd_d(NAN), r);
+
+  o = vand_vo_vo_vo(vand_vo_vo_vo(vor_vo_vo_vo(veq_vo_vd_vd(a, vcast_vd_d(INFINITY)), visnumber_vo_vd(a)),
+				  vge_vo_vd_vd(a, vcast_vd_d(-DBL_MIN))),
+		    vor_vo_vo_vo(vor_vo_vo_vo(veq_vo_vd_vd(a, vcast_vd_d(0)), vgt_vo_vd_vd(a, vcast_vd_d(200))), visnan_vo_vd(r)));
+  r = vsel_vd_vo_vd_vd(o, vmulsign_vd_vd_vd(vcast_vd_d(INFINITY), a), r);
+  
+  return r;
+}
+
+EXPORT CONST vdouble xlgamma_u1(vdouble a) {
+  dd2 d = gammak(a);
+  vdouble2 y = ddadd2_vd2_vd2_vd2(d.a, logk2(ddabs_vd2_vd2(d.b)));
+  vdouble r = vadd_vd_vd_vd(y.x, y.y);
+  vopmask o;
+
+  o = vor_vo_vo_vo(visinf_vo_vd(a),
+		   vor_vo_vo_vo(vand_vo_vo_vo(vle_vo_vd_vd(a, vcast_vd_d(0)), visint_vo_vd(a)),
+				vand_vo_vo_vo(visnumber_vo_vd(a), visnan_vo_vd(r))));
+  r = vsel_vd_vo_vd_vd(o, vcast_vd_d(INFINITY), r);
+
+  return r;
+}
+
+EXPORT CONST vdouble xerf_u1(vdouble a) {
+  vdouble s = a, t, u;
+  vdouble2 d;
+
+  a = vabs_vd_vd(a);
+  vopmask o0 = vlt_vo_vd_vd(a, vcast_vd_d(1.0));
+  vopmask o1 = vlt_vo_vd_vd(a, vcast_vd_d(3.7));
+  vopmask o2 = vlt_vo_vd_vd(a, vcast_vd_d(6.0));
+  u = vsel_vd_vo_vd_vd(o0, vmul_vd_vd_vd(a, a), a);
+  
+  t = vsel_vd_vo_vo_d_d_d(o0, o1, +0.6801072401395392157e-20, +0.2830954522087717660e-13, -0.5846750404269610493e-17);
+  t = vmla_vd_vd_vd_vd(t, u, vsel_vd_vo_vo_d_d_d(o0, o1, -0.2161766247570056391e-18, -0.1509491946179481940e-11, +0.6076691048812607898e-15));
```

Comment:
Same here, please add a TODO comment about the potential optimization that can be done for architectures with indexed `mla` instructions.